component/common/src/main/java/org/wso2/carbon/extension/identity/emailotp/common/dto/SessionDTO.java

```diff
@@ -0,0 +1,135 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.extension.identity.emailotp.common.dto;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Object used to save the data in the database.
+ */
+public class SessionDTO implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private String otpToken;
+    private long generatedTime;
+    private long expiryTime;
+    private String transactionId;
+    private String fullQualifiedUserName;
+    private String userId;
+
+    public String getOtpToken() {
+
+        return otpToken;
+    }
+
+    public void setOtpToken(String otpToken) {
+
+        this.otpToken = otpToken;
+    }
+
+    public long getGeneratedTime() {
+
+        return generatedTime;
+    }
+
+    public void setGeneratedTime(long generatedTime) {
+
+        this.generatedTime = generatedTime;
+    }
+
+    public long getExpiryTime() {
+
+        return expiryTime;
+    }
+
+    public void setExpiryTime(long expiryTime) {
+
+        this.expiryTime = expiryTime;
+    }
+
+    public String getTransactionId() {
+
+        return transactionId;
+    }
+
+    public void setTransactionId(String transactionId) {
+
+        this.transactionId = transactionId;
+    }
+
+    public String getFullQualifiedUserName() {
+
+        return fullQualifiedUserName;
+    }
+
+    public void setFullQualifiedUserName(String fullQualifiedUserName) {
+
+        this.fullQualifiedUserName = fullQualifiedUserName;
+    }
+
+    public String getUserId() {
+
+        return userId;
+    }
+
+    public void setUserId(String userId) {
+
+        this.userId = userId;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof SessionDTO)) {
+            return false;
+        }
+        SessionDTO that = (SessionDTO) o;
+        return getGeneratedTime() == that.getGeneratedTime() &&
+                getExpiryTime() == that.getExpiryTime() &&
+                getOtpToken().equals(that.getOtpToken()) &&
+                getTransactionId().equals(that.getTransactionId()) &&
+                getFullQualifiedUserName().equals(that.getFullQualifiedUserName()) &&
+                getUserId().equals(that.getUserId());
+    }
+
+    @Override
+    public int hashCode() {
+
+        return Objects.hash(getOtpToken(), getGeneratedTime(), getExpiryTime(), getTransactionId(),
+                getFullQualifiedUserName(), getUserId());
+    }
+
+    @Override
+    public String toString() {
+
+        return "SessionDTO{" +
+                "otpToken='" + otpToken + '\'' +
+                ", generatedTime=" + generatedTime +
+                ", expiryTime=" + expiryTime +
+                ", transactionId='" + transactionId + '\'' +
+                ", fullQualifiedUserName='" + fullQualifiedUserName + '\'' +
+                ", userId='" + userId + '\'' +
+                '}';
```

Comment:
Can't we make use of `Gson().toJson()` in places this is called?

Reply:
This is `toString()` method and there is no need to convert string to a `JSON` object.